src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java

```diff
@@ -1118,10 +1129,13 @@ public ValidationResult validate(NamespaceInfo namespace, boolean isNew) {
         if ( existing != null && !existing.getId().equals( namespace.getId() ) ) {
             throw new IllegalArgumentException( "Namespace with prefix '" + namespace.getPrefix() + "' already exists.");
         }
-        
-        existing = getNamespaceByURI( namespace.getURI() );
-        if ( existing != null && !existing.getId().equals( namespace.getId() ) ) {
-            throw new IllegalArgumentException( "Namespace with URI '" + namespace.getURI() + "' already exists.");
+
+        if (!namespace.isIsolated()) {
+            // not an isolated namespace \ workplace so we need to check for duplicates
+            existing = getNamespaceByURI(namespace.getURI());
```

Comment:
Say we are adding a non isolated workspace that "conflicts" with an isolated one, it should be accepted no?. What prevents getNamespaceByURI from returning the isolated one? I checked quickly in IsolatedCatalogFacade and did not see anything that would prevent this issue.

Reply:
Ok, found it in DefaultCatalogFacade... a bit surprising though, wouldn't it make more sense to keep that logic in the IsolatedCatalogFacade wrapper instead? Ideally, having no changes in DefaultCatalogFacade at all, leaving all in the wrapper, would ease plugging other catalog facade implementation by simple wrapping

Reply:
I see... however other facades should be modified as well, it would be a change in expected behavior (so javadocs, migration guide). The other possibility would be to iterate over all namespaces in the wrapper, but I'm not sure what performance effect it would have.

Reply:
Oh, another possibility could be to extend the Facade with a getNamespacesByURI method whose default implementation is a simple iterator, but DefaultCatalogFacade could have a smarter version of it (there should not be many matching the same URI, right?)